ql/src/java/org/apache/hadoop/hive/ql/exec/tez/HiveSplitGenerator.java

```diff
@@ -91,9 +92,16 @@
   private final MapWork work;
   private final SplitGrouper splitGrouper = new SplitGrouper();
   private final SplitLocationProvider splitLocationProvider;
+  private final OptionalInt numSplits;
```

Comment:
Since the constructor accepts an `Integer` type anyway, can you please make this `Optional<Integer>` ?